build/local/Makefile

```diff
@@ -31,6 +31,7 @@ build: setup
 docker-build: setup
 	docker build -t ${IMAGE_NAME}:${IMAGE_VERSION} .
 	sed -i "s/%OTEL_COLLECTOR_IMAGE%/${REGISTRY_LOCATION}-docker.pkg.dev\/${GCLOUD_PROJECT}\/${CONTAINER_REGISTRY}\/${IMAGE_NAME}:${IMAGE_VERSION}/g" ../../deploy/gke/simple/manifest.yaml
+	sed -i "s/%OTEL_COLLECTOR_IMAGE%/${REGISTRY_LOCATION}-docker.pkg.dev\/${GCLOUD_PROJECT}\/${CONTAINER_REGISTRY}\/${IMAGE_NAME}:${IMAGE_VERSION}/g" ../../deploy/gke/redaction/manifest.yaml
```

Comment:
I see, makes sense. I initially thought adding a line in the Readme about how to edit this file, but found that approach to be potentially confusing.  

What should our intent be here - 
 -  Do we need a better way to achieve what this build target is doing ? 
  OR
 -  Do we need to find an alternative to this approach altogether ? (Something along the lines to find a way to specify what sample to build the image for)

Reply:
A better (more generic) way to do what this build target is doing. For example, something like a common target that each sample's Makefile imports and the sample sets which files to update is the kind of approach I'm thinking of

Reply:
This gives me an idea, if we apply the sed command in each sample's makefile and ask the user to run that target before deploying the sample, then that would also fix this problem. 

A good side benefit is that in case the collector image is already built and pushed to Artifact Registry, the users would not have to `make docker-build` again just to update the manifest files with the correct names. This also kind of decouples the `build`  and `deploy` directories since running commands from the build directory would not edit the files in deploy.